spec/unit/inventory/inventory_spec.rb

```diff
@@ -285,11 +285,26 @@ def get_target(inventory, name, alia = nil)
           expect(targets).to eq(matched_targets)
         end
 
-        it 'should match wildcard selectors' do
+        it 'should match wildcard selector *' do
           targets = inventory.get_targets('target*')
           expect(targets.map(&:name).sort).to eq(%w[target1 target2 target3 target4 target5 target6 target7 target9])
         end
 
+        it 'should match wildcard selector ?' do
+          targets = inventory.get_targets('target?')
+          expect(targets.map(&:name).sort).to eq(%w[target1 target2 target3 target4 target5 target6 target7 target9])
+        end
+
+        it 'should match wildcard selector []' do
+          targets = inventory.get_targets('target[0-9]')
+          expect(targets.map(&:name).sort).to eq(%w[target1 target2 target3 target4 target5 target6 target7 target9])
+        end
+
+        it 'should match wildcard selector {}' do
+          targets = inventory.get_targets('{target,node}[0-9]')
+          expect(targets.map(&:name).sort).to eq(%w[target1 target2 target3 target4 target5 target6 target7 target9])
+        end
```

Comment:
This test doesn't confirm that using the `{}` wildcard selector works, since the inventory doesn't include any targets named `node[0-9]`. I'd add at least one node with a matching name to the inventory (starting on line 104) and update this test to include that node in the expected list.

I'd also add a test that uses a combination of glob patterns with commas and target names (e.g. `{target,node}[1-3],target9` or similar) to test that the regex on line 150 works as expected.